pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -451,27 +417,137 @@ func (b *bootstrapper) podLabels() map[string]string {
 	return nil
 }
 
-func (b *bootstrapper) containerCommand() []string {
+func (b *bootstrapper) getContainerCommand() []string {
 	switch b.values.Role {
-	case RoleEndpoint:
+	case RoleWeeder:
 		return []string{
 			"/usr/local/bin/dependency-watchdog",
+			"weeder",
 			fmt.Sprintf("--config-file=%s/%s", volumeMountPath, configFileName),
-			"--deployed-namespace=" + b.namespace,
-			"--watch-duration=5m",
+			"--enable-leader-election=true",
 		}
 
-	case RoleProbe:
+	case RoleProber:
 		return []string{
 			"/usr/local/bin/dependency-watchdog",
-			"probe",
+			"prober",
 			fmt.Sprintf("--config-file=%s/%s", volumeMountPath, configFileName),
-			"--deployed-namespace=" + b.namespace,
-			"--qps=20.0",
-			"--burst=100",
-			"--v=4",
+			"--kube-api-qps=20.0",
+			"--kube-api-burst=100",
+			"--zap-log-level=INFO",
+			"--enable-leader-election=true",
 		}
 	}
 
 	return nil
 }
+
+func (b *bootstrapper) getDeployment(serviceAccountName string, configMapName string) *appsv1.Deployment {
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.name(),
+			Namespace: b.namespace,
+			Labels: utils.MergeStringMaps(map[string]string{
+				resourcesv1alpha1.HighAvailabilityConfigType: resourcesv1alpha1.HighAvailabilityConfigTypeController,
+			}, b.getLabels()),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas:             pointer.Int32(1),
+			RevisionHistoryLimit: pointer.Int32(2),
+			Selector:             &metav1.LabelSelector{MatchLabels: b.getLabels()},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: b.getPodLabels(),
```

Comment:
When rebasing, make sure to include the changes from https://github.com/gardener/gardener/pull/7636.